### ISSUE TITLE
Save branch as scene saves children without owner set

### ISSUE THREAD

**Issue by marcinn1** [godotengine/godot#82756]
### Godot version

4.2.dev [44e399ed5fa895f760b2995e59788bdb49782666]

### System information

Linux

### Issue description

I have node with a `@tool` script. The script adds child nodes needed only for the editor.

![savebranch_bug_1](https://github.com/godotengine/godot/assets/133269979/dfaab1e3-e125-4b60-b0e3-72324b3a93b9)

I'm saving the branch of `CustomNode` as a new scene, and the CSGBox saves too, which is unexpected behaviour:

![savebranch_bug_2](https://github.com/godotengine/godot/assets/133269979/c266bbc6-25d3-4775-8f51-846f0618a1dc)

### Steps to reproduce

- create a new scene
- add node
- attach script:
```
@tool
extends Node3D

var _box = CSGBox3D.new()

func _enter_tree():
    add_child(_box)
    
func _remove_tree():
    remove_child(_box)
```
- save added node as a new scene
- open saved scene

### Minimal reproduction project

[save_branch_bug.zip](https://github.com/godotengine/godot/files/12797975/save_branch_bug.zip)

1. open the project
2. open `save_branch_bug.tscn` if it is not opened already
3. RMB on the `CustomNode` and select "Save branch as a new scene"
4. Open saved scene

You will see CSGBox in a nodes tree. Expected result: no CSGBox node visible in the tree. To make sure: click on the `_CSGBox3D_<number>` node and change it's position. You will notice two boxes. Should be just one.

**Comment by KoBeWi** [comment]
#61966 attempted to fix a similar issue. The question is whether we want to handle unowned nodes in the editor (like it was before #45937) or it's fine to have an exception, as saving branch as scene is not the same as using `pack()`. If we go with the latter then you are supposed to use internal children if you don't want them saved.

**Comment by marcinn1** [comment]
> The question is whether we want to handle unowned nodes in the editor or it's fine to have an exceptio

Hi. I will always be in favor of consistency and predictability, mostly in terms of UI/UX (i.e. workflow), not only consistency of `pack()`. You have a possibility to make instanced branch local ("Make local"), and also you can save branch as a separate scene ("Save branch..."). This procedure should be reversible, i.e. saving branch as scene, then making it local again should give same result as before (and vice versa). This was possible before 0559fc58d17c82983bfa6d37207642178932ebde

**Comment by marcinn1** [comment]
> I was going to suggest using internal children, but they seem bugged for whatever reason

Hmm, I'll check this again. I haven't tried adding tool nodes as internal.

Regardless, please bear in mind that subtree can be unexpectedly modified by adding unowned nodes when saving branch as a scene, i.e. the workflow "local node <-> saved branch" is not reversible anymore. This can affect other use cases. If someone really wants to save all nodes, including ghosts/unowned, I would suggest adding an "Include unowned children" checkbox somewhere, disabled by default. Disabled, because most users wants to save what they see (-> WYSIWYG).

**Comment by KoBeWi** [comment]
I don't think it was intended. The regression was fixed afterwards, but the fixed behavior was still different.

**Comment by marcinn1** [comment]
Thanks.

I'm attaching the patch I'm currently testing. Maybe you or someone else would like to check it against regressions. It just reverts 0559fc58d17c82983bfa6d37207642178932ebde, so I'm not too happy with that. But most important for now is to fix the regression.

[0001-Revert-of-0559fc5-as-potential-fix-for-82756.patch.txt](https://github.com/godotengine/godot/files/12803739/0001-Revert-of-0559fc5-as-potential-fix-for-82756.patch.txt)

**Comment by KoBeWi** [comment]
Reverting that will most likely re-introduce #42611, so it's not acceptable.

**Comment by marcinn1** [comment]
@KoBeWi There is unused `duplimap` which comes from 0559fc58d17c82983bfa6d37207642178932ebde. The solution is to use `duplimap` and check that the original nodes had an owner set.

**Comment by Rindbee** [comment]
It seems that the original node can be referenced to save the branch, because the nodes visible in the scene tree are all nodes with an owner. What we want to save are these visible nodes. And their owners should have the same owner as the new scene root.

**Comment by marcinn1** [comment]
Godot currently saves not only visible nodes in scene tree. It saves unvisible / unowned, too (i.e. added by editor tools). This is unexpected.

**Comment by Rindbee** [comment]
> Godot currently saves not only visible nodes in scene tree. It saves unvisible / unowned, too (i.e. added by editor tools). This is unexpected.

The solution is to take advantage of the fact that they are not visible in the scene dock and exclude them.

**Comment by marcinn1** [comment]
Yes. And you can do that by checking ownership of the original node (before duplication). They're not a subscenes yet, and the logic of saving a branch is not using `_add_node`.

https://github.com/godotengine/godot/pull/82802/commits/ae3a47781845341bb160dec15fa578a07bf25713
